superset/views/core.py

```diff
@@ -1958,11 +1958,26 @@ def extra_table_metadata(self, database_id, table_name, schema):
     @expose("/select_star/<database_id>/<table_name>/<schema>")
     @event_logger.log_this
     def select_star(self, database_id, table_name, schema=None):
-        mydb = db.session.query(models.Database).get(database_id)
+        logging.warning(
+            "This API endpoint is deprecated and will be removed in version 1.0.0"
```

Comment:
Let's include the actual endpoint being deprecated in the warning - I imagine there will be a number deprecated for 1.0 and it would be helpful for users not to have to dig through the logs to understand which endpoint is issuing the warning.

Reply:
Of course can do that. Was assuming the use of custom logging setup that can populate function names, line numbers etc...

Reply:
one other thing that might be nice, is adding a statsd counter for each deprecated function, so that admins can easily check those counters to see if they're using any deprecated functions

Reply:
ok, good idea @etr2460

Reply:
I've added the following metrics:
- init_Superset.select_star
- database_not_found_Superset.select_star
- permission_denied_Superset.select_star
- success_Superset.select_star

- init_DatabaseRestApi.select_star
- database_not_found_DatabaseRestApi.select_star
- permission_denied_DatabaseRestApi.select_star
- success_DatabaseRestApi.select_star

- init_DatabaseRestApi.table
- database_not_found_DatabaseRestApi.table
- permission_denied_DatabaseRestApi.table
- success_DatabaseRestApi.table

Reply:
Thanks @dpgaspar! do you think these should be prefixed by `deprecated_function_call` or something like that?

Reply:
Could be useful on your side. So I've made a few changes (following @robdiciuccio comment):

metrics are now:

deprecated.Superset.select_star.init

deprecated.Superset.select_star.database_not_found

deprecated.Superset.select_star.permission_denied

deprecated.Superset.select_star.success

DatabaseRestApi.select_star.init

DatabaseRestApi.select_star.database_not_found

DatabaseRestApi.select_star.permission_denied

DatabaseRestApi.select_star.success

DatabaseRestApi.table_metada.init

DatabaseRestApi.table_metada.database_not_found

DatabaseRestApi.table_metada.permission_denied

DatabaseRestApi.table_metada.success